dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -288,154 +279,394 @@ def read_metadata(
-        # if we have a list of files and gather_statistics=False
-        base_path = (base_path or "") + fs.sep
-        if parts:
-            # Case (1)
-            # `parts` is just a list of path names.
-            pqpartitions = None
-            pf_deps = None
-            partsin = parts
-        else:
-            # Populate `partsin` with dataset row-groups
-            partsin = pf.row_groups
-            pqpartitions = pf.info.get("partitions", None)
-            if pqpartitions and not fast_metadata:
-                # Case (2)
-                # We have parquet partitions, and do not have
-                # a "_metadata" file for the worker to read.
-                # Therefore, we need to pass the pf object in
-                # the task graph
-                pf_deps = pf
-            else:
-                # Case (3)
-                # We don't need to pass a pf object in the task graph.
-                # Instead, we can try to pass the path for each part.
-                pf_deps = "tuple"
-
-        parts = []
-        i_path = 0
-        path_last = None
-
-        # Loop over DataFrame partitions.
-        # Each `part` will be a row-group or path ()
-        for i, part in enumerate(partsin):
-            if pqpartitions and fast_metadata:
-                # Case (3A)
-                # We can pass a "_metadata" path
-                file_path = base_path + "_metadata"
-                i_path = i
-            elif (
-                pf_deps
-                and isinstance(part.columns[0].file_path, str)
-                and not pqpartitions
+        return meta, dtypes, index_cols, categories_dict, categories, index
+
+    @classmethod
+    def _update_metadata_options(
+        cls,
+        pf,
+        parts,
+        gather_statistics,
+        split_row_groups,
+        index_cols,
+        filters,
+    ):
+        # Cannot gather_statistics if our `parts` is alreadey a list
+        # of paths, or if we are building a multiindex (for now).
+        # We also don't "need" to gather statistics if we don't
+        # want to apply any filters or calculate divisions.
+        if (
+            isinstance(parts, list) and len(parts) and isinstance(parts[0], str)
+        ) or len(index_cols) > 1:
+            gather_statistics = False
+        elif filters is None and len(index_cols) == 0:
+            gather_statistics = False
+
+        # Determine which columns need statistics.
+        flat_filters = _flatten_filters(filters)
+        stat_col_indices = {}
+        for i, name in enumerate(pf.columns):
+            if name in index_cols or name in flat_filters:
+                stat_col_indices[name] = i
+
+        # If the user has not specified `gather_statistics`,
+        # we will only do so if there are specific columns in
+        # need of statistics.
+        if gather_statistics is None:
+            gather_statistics = len(stat_col_indices.keys()) > 0
```

Comment:
This is a really good question.  Right now, there is no reason at all to allow `gather_statistics=True` if `stat_col_indices` is empty, because we will still be using `stat_col_indices` to choose which columns to gather statistics for.   However, this may become a problem for implementing "predicate pushdown" in the future.  I think it is fine to cross that bridge when we get there :)

Reply:
It looks like we cannot change `gather_statistics` from True to False (even if `stat_col_indices` is empty), in case a `chunksize` is specified.  In the `chunksize` case, the column-chunk statistics will not be collected, but the row-group statistics are still needed to aggregate the elements of `parts`.

I'll add another note about this.